ToolChain/Assembler/Parser/test_Parser.py

```diff
@@ -37,1824 +35,1398 @@
 
 class TestParser(unittest.TestCase):
 
-    parser = Parser(skipValidation=True)
+    parser = Parser()
+
+    def test_parse(self):
+        """
+        Tests the method:
+        build(self, text)
+        """
+
+        self.parser.relativeAddressCounter = 0
+        text = "MOV $B $C"
+        instruction, relAddress, flag = self.parser.parse(text)
+        self.assertEqual(instruction, bytes((0b10011011,)) + bytes((0b00010010,)))
+        self.assertEqual(relAddress, 2)
+        self.assertEqual(flag, 0)
+
+        self.parser.relativeAddressCounter = 0
+        text = ".dataAlpha test string  two  spaces\n"
+        instruction, relAddress, flag = self.parser.parse(text)
+        self.assertEqual(instruction, b'test string  two  spaces\x00')
+        self.assertEqual(relAddress, 25)
+        self.assertEqual(flag, 0)
+
+        self.parser.relativeAddressCounter = 0
+        text = ".global start"
+        instruction, relAddress, flag = self.parser.parse(text)
+        self.assertEqual(instruction, "START")
+        self.assertEqual(relAddress, 0)
+        self.assertEqual(flag, 2)
+
+        self.parser.relativeAddressCounter = 0
+        text = "start:"
+        instruction, relAddress, flag = self.parser.parse(text)
+        self.assertEqual(instruction, "START")
+        self.assertEqual(relAddress, 0)
+        self.assertEqual(flag, 1)
 
-    def test_init(self):
+        self.parser.relativeAddressCounter = 0
+        text = "MOV start $B"
+        instruction, relAddress, flag = self.parser.parse(text)
+        self.assertEqual(instruction, bytes((0b01100000,)) + b':START:' + bytes((0b00000001,)))
+        self.assertEqual(relAddress, 6)
+        self.assertEqual(flag, 0)
+
+    def test_parseError(self):
         """
-        Test the method:
-        __init__(self, file: str="")
+        Tests various errors for the method:
+        build(self, text)
         """
-        self.assertRaises(ValueError, Parser)
 
+        try:
```

Comment:
There is a cleaner way of doing this. Look at examples in :
https://github.com/CommunicationsSecurityEstablishment/spartacus/blob/be4504a020b061e67dd9fa26fe386aee4ad01673/CapuaEnvironment/ExecutionUnit/test_ExecutionUnit.py#L54

Reply:
Same is true for every parts where a try/except is used for exception validation.